doc/design/design_high-available_sqlflow_server.md

```diff
@@ -0,0 +1,231 @@
+# High-available SQLFlow Server
+
+## Motivations
+
+In the current system, the SQLFlow client connects the SQLFlow server with a long live connection.
+the SQLFlow client sends a gRPC request which contains a SQL statement and wait until the SQLFlow server finishes executing the SQL statement.
+
+When the SQLFlow server receives one training SQL statement, it will generate a python training program that runs on the host or submit the training job to some distributed training service cluster (Kubernetes/Yarn). This will cause:
```

Comment:
Just say submit a job, not to mention running on the host, which means submit jobs to run on the host. No need to mention Kubernetes and YARN either.